core/src/main/java/com/graphhopper/routing/util/countryrules/europe/BelarusCountryRule.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper.routing.util.countryrules.europe;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.ev.RoadClass;
+import com.graphhopper.routing.ev.Toll;
+import com.graphhopper.routing.util.TransportationMode;
+import com.graphhopper.routing.util.countryrules.CountryRule;
+
+public class BelarusCountryRule implements CountryRule {
+    
+    @Override
+    public Toll getToll(ReaderWay readerWay, TransportationMode transportationMode, Toll currentToll) {
+        if (!transportationMode.isMotorVehicle() || currentToll != Toll.MISSING) {
+            return currentToll;
+        }
+
+        RoadClass roadClass = RoadClass.find(readerWay.getTag("highway", ""));
```

Comment:
Sooner or later I think the country rules should have available the encoded values that were written already at this point so we could just get the road class as encoded value rather than repeating the road class parsing here. We would probably just add the encoding manager to the country rules via their constructor. Whether or not this will work will depend on the road class parser being executed before the toll parser, but this is exactly what we were planning to do (see #2470): The order of the parsers matters and allows encoded values to depend on others.

Again, since we are adding so many countries here, can we make this change already?

Reply:
So we would need to add dependency resolution?

```java
public interface TagParser {
    
    Set<String> getProvidedEncodedValues();
    
    default Set<String> getEncodedValueDependencies() {
        return Collections.emptySet();
    }
```

We could do a topological sort within `EncodingManager.Builder.build()`. With the parsers in the right order, we can safely pass the EncodedValueLookup into the CountryRules.

Reply:
Hm, not sure the original idea was just doing this manually, i.e. the order of the list of tag parsers counts. But actually if all you need to do here is reading the highway tag, maybe let's postpone this.